extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -286,11 +333,29 @@ export class Bind {
     // Helper function for scanning the tree walker's next node.
     // Returns true if the walker has no more nodes.
     const scanNextNode_ = () => {
-      const element = walker.nextNode();
-      if (!element) {
+      const node = walker.currentNode;
+      if (!node) {
         return true;
       }
+      // Walker is filtered to only return elements
+      const element = dev().assertElement(node);
       const tagName = element.tagName;
+      if (DYNAMIC_TAGS.includes(tagName)) {
+        // Listen for changes in amp-mustache templates
+        // Templated HTML is added as a sibling to the template tag.
+        // So observe the parent.
+        // TODO(kmh287): What if parent is the body tag?
+        // TODO(kmh287): Generify logic for node observation strategy
+        // when bind supprots more dynamic nodes.
+        const elementToObserve = tagName === 'TEMPLATE' ?
+          element.parentElement :
+          element;
+        this.subtreeMutationObserver_.observe(elementToObserve, {
+          childList: true,
+        });
+        this.dynamicRoots_.push(elementToObserve);
+      }
```

Comment:
Can you extract this code block into a separate function? Also, just keep it simple for now and extend the functionality later (just check for TEMPLATE tag). It prevents readability degradation in the interim of undetermined length.